otherlibs/dune-rpc/dune_rpc.mli

```diff
@@ -254,7 +254,8 @@ module V1 : sig
         -> 'a
         -> ('b, Response.Error.t) result fiber
 
-      val notification : t -> 'a Notification.t -> 'a -> unit fiber
+      val notification :
+        t -> 'a Notification.t -> 'a -> (unit, Response.Error.t) result fiber
```

Comment:
If notifications are returning a result? Why do we need them at all? What's the difference between a notification and request then?

Reply:
I wasn't sure how to express this pattern (maybe we could swallow versioning errors?); it's kind of messy.

Contrary to the implied type, notifications do not wait for a response from the server. The error variant is only produced in the case that we try to send a notification that the server does not understand (due to mismatched versions). I agree that this is confusing; I'd appreciate any better ideas.

Reply:
I think we can work around this in two ways:

1. Change the return type to be `(unit Fiber.t, Response.Error.t) result`. This reflects the fact that error handling is done on the client side.

2. Make the prepare step explicit.
```ocaml
val 'a notification
val prepare_notification : Client.t -> 'a Notification.t -> ('a notification, error) result
val notification : 'a Client.t -> 'a notification -> 'a -> unit fiber
```

Reply:
As discussed on slack, we just need a better error type for this function and this PR is good to go.